src/com/esotericsoftware/kryo/Kryo.java

```diff
@@ -1161,22 +1161,15 @@ public boolean isClousre (Class type) {
 
 	public void pushGenericsScope (Class type, Generics generics) {
 		if (TRACE) trace("kryo", "Settting a new generics scope for class " + type.getName() + ": " + generics);
-		Generics currentScope = genericsScope;
-		if (generics.getParentScope() != null) {
-			generics = new Generics(generics.getMappings());
-		}
-		genericsScope = generics;
-		genericsScope.setParentScope(currentScope);
+		genericsResolver.pushScope(generics);
 	}
 
 	public void popGenericsScope () {
-		Generics oldScope = genericsScope;
-		if (genericsScope != null) genericsScope = genericsScope.getParentScope();
-		if (oldScope != null) oldScope.resetParentScope();
+		genericsResolver.popScope();
 	}
 
-	public Generics getGenericsScope () {
-		return genericsScope;
+	public GenericsResolver getGenericsScope () {

```

Comment:
I'm a bit unhappy about the fact that `getGenericScope` returns a value of type `GenericsResolver`. I think it would be better if you'd rename the method into `getGenericResolver`, similar to what we do for other resolvers in Kryo.